packages/core/src/api/clipboard/fromClipboard/pasteExtension.ts

```diff
@@ -26,6 +30,25 @@ function defaultPasteHandler({
   prioritizeMarkdownOverHTML: boolean;
   plainTextAsMarkdown: boolean;
 }) {
+  // Special case for code blocks, as they do not support any rich text
+  // formatting, so we force pasting plain text.
+  const selection = editor.prosemirrorView?.state.selection;
```

Comment:
This API is discouraged now with the introduction of BlockNote transactions, you can instead do this:

```ts
  // Special case for code blocks, as they do not support any rich text
  // formatting, so we force pasting plain text.
  const isInCodeBlock = editor.transact((tr) => {
    const blockInfo = getBlockInfoFromTransaction(tr);

    const selectionInCodeBlock =
      blockInfo.isBlockContainer && blockInfo.blockContent.node.type.spec.code;

    return selectionInCodeBlock;
  });

  if (isInCodeBlock) {
    const data = event.clipboardData?.getData("text/plain");
    if (data) {
      editor.pasteText(data);
      return true;
    }
  }
```